### ISSUE TITLE
[proposal] add `const` at compile time (where possible)

### ISSUE THREAD

**Issue by iapicca** [dart-lang/sdk#56669]
#### related issues
- https://github.com/dart-lang/lints/issues/205 
- https://github.com/flutter/flutter/issues/149932

note quote from the former
> In theory, const-ness should give apps a performance boost, but during development the lints enforcing const are constantly nagging developers: "Make this const" or "This cannot be const anymore".

### proposal

if a linter is able to tell if a variable, a constructor or (in flutter) a child **_should_** be a constant, 
wouldn't be more developer friendly to automatically make it a constant at compile time
without ([quote](https://github.com/flutter/flutter/issues/149932#issue-2341109065)) "nagging developers"

**Comment by mateusfccp** [comment]
The issue https://github.com/dart-lang/language/issues/1410 has more discussion about the implications of this request.

**Comment by rakudrama** [comment]
Another approach is to change the language to have more contexts that are `const` if possible.
See https://github.com/dart-lang/language/issues/3399

TL;DR:
> The lints have quick-fixes to help with this process, but pushing `const` around the code would be _completely unnecessary_ if the widget constructors defaulted to `const` whenever possible.
